app/scripts/modules/openstack/serverGroup/details/securityGroup/editSecurityGroups.modal.controller.js

```diff
@@ -0,0 +1,91 @@
+  require('../../../../core/task/taskExecutor.js'),
+])
+  .controller('OpenstackEditSecurityGroupsCtrl', function($scope, $uibModalInstance, taskMonitorService, taskExecutor, _,
+                                                 serverGroupWriter, securityGroupReader,
+                                                 application, serverGroup, securityGroups) {
+
+    _.assign($scope, {
+      application: application,
+      filter: {
+        account: serverGroup.account,
+        region: serverGroup.region
+      },
+    });
+
+    var ctrl = this;
+    _.assign(ctrl, {
+      serverGroup: serverGroup,
+      securityGroups: securityGroups.slice(0).sort((a, b) => a.name.localeCompare(b.name)),
+      state: { verification: {} },
+      isValid: () => { return ctrl.securityGroups.length > 0; },
+      submit: () => {
+        var submitMethod = () => {
+          return serverGroupWriter.updateSecurityGroups(
+            serverGroup,
+//            _.map(ctrl.securityGroups, (sg) => { return { id: sg }; }),
+            ctrl.securityGroups,
+            application);
+        };
+
+        var taskMonitorConfig = {
+          modalInstance: $uibModalInstance,
+          application: application,
+          title: 'Update Security Groups for ' + serverGroup.name,
+          onTaskComplete: application.serverGroups.refresh,
+        };
+
+        ctrl.taskMonitor = taskMonitorService.buildTaskMonitor(taskMonitorConfig);
+        ctrl.taskMonitor.submit(submitMethod);
+      },
+      cancel: $uibModalInstance.dismiss
+    });
+
+//    this.isValid = () => this.state.verification.verified;
+
+//    securityGroupReader.getAllSecurityGroups().then(allGroups => {
+//      let account = serverGroup.account,
+//          region = serverGroup.region,
+//          vpcId = serverGroup.vpcId;
+//      this.availableSecurityGroups = _.get(allGroups, [account, 'openstack', region].join('.'), []);
+//      this.state.securityGroupsLoaded = true;
+//
+//      var chosen = {};
+//      _.forEach(this.command.securityGroups, function(sg) {
+//        chosen[sg.id] = true;
+//      });
+//      this.command.securityGroups = _.filter(this.availableSecurityGroups, function(sg) {
+//        return chosen[sg.id];
+//      });
+//    });
+
+//    this.serverGroup = serverGroup;
+
+//    this.submit = () => {
+//      var submitMethod = () => {
+//        this.state.submitting = true;
+//        return serverGroupWriter.updateSecurityGroups(serverGroup, this.command.securityGroups, application);
+//      };
+//
+//      var taskMonitorConfig = {
+//        modalInstance: $uibModalInstance,
+//        application: application,
+//        title: 'Update Security Groups for ' + serverGroup.name,
+//        onTaskComplete: application.serverGroups.refresh,
+//      };
+//
+//      this.taskMonitor = taskMonitorService.buildTaskMonitor(taskMonitorConfig);
+//
+//      this.taskMonitor.submit(submitMethod);
+//    };
+//
+//    this.cancel = $uibModalInstance.dismiss;
```

Comment:
I'll take the files out of the commit... there was some work that would be useful if we later implement edit of security groups for server groups... but it is probably best not to carry around such dead code.